big_tests/tests/extdisco_SUITE.erl

```diff
@@ -0,0 +1,338 @@
+        Iq = request_external_services_incorrect(domain()),
+        escalus_client:send(Alice, Iq),
+
+        % Then empty iq_result is returned
+        Result = escalus_client:wait_for_stanza(Alice),
+        escalus:assert(is_iq_result, [Iq], Result),
+        ?assertEqual(false, has_subelement_with_ns(Result, <<"incorrect">>, ?EXT_DISCO))
+    end,
+    escalus:fresh_story(Config, [{alice, 1}], Test).
+
+%%-----------------------------------------------------------------
+%% Helpers
+%%-----------------------------------------------------------------
+
+domain() ->
+    ct:get_config({hosts, mim, domain}).
+
+set_external_services(Opts, C) ->
+    Module = [{mod_extdisco, Opts}],
+    Config = dynamic_modules:save_modules(domain(), C),
+    ok = dynamic_modules:ensure_modules(domain(), Module),
+    Config.
+
+remove_external_services(C) ->
+    OldModules = rpc(mim(), gen_mod, loaded_modules_with_opts, [domain()]),
+    NewModules = lists:keydelete(mod_extdisco, 1, OldModules),
+    ok = rpc(mim(), gen_mod_deps, replace_modules, [domain(), OldModules, NewModules]),
+    C.
+
+request_external_services(To) ->
+    escalus_stanza:iq(To, <<"get">>,
+        [#xmlel{name = <<"services">>,
+                attrs = [{<<"xmlns">>, <<"urn:xmpp:extdisco:2">>}]}]).
+
+request_external_services_with_type(To, Type) ->
+    escalus_stanza:iq(To, <<"get">>,
+        [#xmlel{name = <<"services">>,
+                attrs = [{<<"xmlns">>, <<"urn:xmpp:extdisco:2">>},
+                         {<<"type">>, Type}]}]).
+
+request_external_services_credentials(To, Type, Host) ->
+    escalus_stanza:iq(To, <<"get">>,
+        [#xmlel{name = <<"credentials">>,
+                attrs = [{<<"xmlns">>, <<"urn:xmpp:extdisco:2">>}],
+                children = [#xmlel{name = <<"service">>,
+                                   attrs = [{<<"host">>, Host},
+                                            {<<"type">>, Type}]}]}]).
+
+request_external_services_incorrect(To) ->
+    escalus_stanza:iq(To, <<"get">>,
+        [#xmlel{name = <<"incorrect">>,
+                attrs = [{<<"xmlns">>, <<"urn:xmpp:extdisco:2">>}]}]).
+
+get_service_element(Result) ->
+    Services = exml_query:subelement_with_ns(Result, ?EXT_DISCO),
+    exml_query:subelements(Services, <<"service">>).
+
+supported_elements() ->
+    [<<"host">>, <<"type">>, <<"port">>, <<"username">>, <<"password">>].
+
+all_services_are_returned(Service) ->
+    [?assertEqual(true, has_subelement(Service, E)) || E <- supported_elements()].
+
+all_services_are_returned(Service, Type) ->
+    ?assertEqual(true, has_attr_with_value(Service, <<"type">>, Type)),
+    all_services_are_returned(Service).
+
+no_services_are_returned(Service) ->
+    [?assertEqual(false, has_subelement(Service, E)) || E <- supported_elements()].
+
+has_subelement(Stanza, Element) ->
+    undefined =/= exml_query:attr(Stanza, Element).
+
+has_attr_with_value(Stanza, Element, Value) ->
+    Value ==  exml_query:attr(Stanza, Element).
```

Comment:
```suggestion
    Value == exml_query:attr(Stanza, Element).
```

Reply:
Why do you use `==` here and `=/=` above and below?

Reply:
This one checks if the attribute has specific value while for the other ones are checking if they have such subelement. For checks if such subelement is there, checking if the return value is not undefined or is not an empty list is enough.